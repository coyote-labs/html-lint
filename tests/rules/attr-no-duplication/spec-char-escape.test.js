const {htmlLint} = require('../../../index');

global.console = {
  error: jest.fn()
};

test.each([
  'tests/rules/attr-no-duplication/error.html',
])(
  'attr-no-duplication %s',
  async(fixture) => {
    try {
      await htmlLint(fixture);
    } catch ({ message }) {
      expect(global.console.error).toMatchSnapshot();
    }
  },
);
