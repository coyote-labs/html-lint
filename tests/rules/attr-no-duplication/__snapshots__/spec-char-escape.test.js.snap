// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attr-no-duplication tests/rules/attr-no-duplication/error.html 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "[101m[html-lint] 2 errors in tests/rules/attr-no-duplication/error.html[49m",
    ],
    Array [
      "[90mattr-no-duplication: Attributes should not be duplicated.[39m
[0m [90m  5 | [39m  [33m<[39m[33m/[39m[33mhead[39m[33m>[39m[0m
[0m [90m  6 | [39m  [33m<[39m[33mbody[39m[33m>[39m[0m
[0m[31m[1m>[22m[39m[90m  7 | [39m    [33m<[39m[33mmain[39m [36mclass[39m[33m=[39m[32m\\"class-one\\"[39m id[33m=[39m[32m\\"id-one\\"[39m id[33m=[39m[32m\\"id-two\\"[39m[33m>[39m[0m
[0m [90m    | [39m                                        [31m[1m^[22m[39m [31m[1mAttributes should not be duplicated.[22m[39m[0m
[0m [90m  8 | [39m      [33m<[39m[33mp[39m [36mclass[39m[33m=[39m[32m\\"hello\\"[39m[33m>[39m[0m
[0m [90m  9 | [39m        [33mHello[39m[33m<[39m[33mspan[39m [36mclass[39m[33m=[39m[32m\\"another class\\"[39m data[33m-[39mtest[33m=[39m[32m\\"light-text\\"[39m data[33m-[39mtest[33m=[39m[32m\\"light-text\\"[39m[33m>[39m[33m,[39m world[33m![39m[33m<[39m[33m/[39m[33mspan[39m[33m>[39m[0m
[0m [90m 10 | [39m      [33m<[39m[33m/[39m[33mp[39m[33m>[39m[0m",
      "
",
    ],
    Array [
      "[90mattr-no-duplication: Attributes should not be duplicated.[39m
[0m [90m  7 | [39m    [33m<[39m[33mmain[39m [36mclass[39m[33m=[39m[32m\\"class-one\\"[39m id[33m=[39m[32m\\"id-one\\"[39m id[33m=[39m[32m\\"id-two\\"[39m[33m>[39m[0m
[0m [90m  8 | [39m      [33m<[39m[33mp[39m [36mclass[39m[33m=[39m[32m\\"hello\\"[39m[33m>[39m[0m
[0m[31m[1m>[22m[39m[90m  9 | [39m        [33mHello[39m[33m<[39m[33mspan[39m [36mclass[39m[33m=[39m[32m\\"another class\\"[39m data[33m-[39mtest[33m=[39m[32m\\"light-text\\"[39m data[33m-[39mtest[33m=[39m[32m\\"light-text\\"[39m[33m>[39m[33m,[39m world[33m![39m[33m<[39m[33m/[39m[33mspan[39m[33m>[39m[0m
[0m [90m    | [39m                                                                [31m[1m^[22m[39m [31m[1mAttributes should not be duplicated.[22m[39m[0m
[0m [90m 10 | [39m      [33m<[39m[33m/[39m[33mp[39m[33m>[39m[0m
[0m [90m 11 | [39m    [33m<[39m[33m/[39m[33mmain[39m[33m>[39m[0m
[0m [90m 12 | [39m  [33m<[39m[33m/[39m[33mbody[39m[33m>[39m[0m",
      "
",
    ],
    Array [
      "
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
